tensorboard/webapp/plugins/npmi/util/umap.ts

```diff
@@ -0,0 +1,113 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {UMAP} from 'umap-js';
+import * as util from '../../../../plugins/projector/vz_projector/util';
+
+const UMAP_SAMPLE_SIZE = 20000;
+
+export interface EmbeddingDataPoint {
+  vector: number[];
+  name: string;
+  sequenceIndex?: number;
+  index: number;
+  projections: {
+    umap?: boolean;
+    'umap-0'?: number;
+    'umap-1'?: number;
```

Comment:
Just wondering, any reason why you are mixing data type in this structure? How about:
```
interface ... {
  ...,
  projections: {
    umap?: Record<number, number>;
  }
}
```

instead?